src/basic/inc/BoxSubset.h

```diff
@@ -66,6 +66,9 @@ class BoxSubset : public VectorSubset<V,M> {
    * m_minValues, and 2) all all components in \c vec are smaller than m_maxValues. */
   bool contains (const V& vec)     const;
 
+  //! Returns the centroid of this box subset.
```

Comment:
Not sure how precise you want to be in these doxygen comments.  The method doesn't actually return anything, it fills up `vec`.